symplyphysics/core/symbols/symbols.py

```diff
@@ -27,23 +27,24 @@ def display_name(self) -> str:
         return self._display_name
 
 
-class DimensionSymbolNew:
+class HasDimension:
```

Comment:
I don't like this approach. You can simply use a Dimension inside your class. HasDimension only introduces `dimension` property which looks too small for a separate class.

Reply:
I see what you mean. I guess my concern was mostly connected to the type system, I thought that [this line](https://github.com/blackyblack/symplyphysics/blob/3d19f392d740f648961085483d3466e411c3981d/symplyphysics/core/quantity_decorator.py#L19) could be reduced (after getting rid of old `DimensionSymbol`) to basically `ScalarValue | SymQuantity | HasDimension` and the next line to `Dimension | HasDimension`, since it's logical that any objects that has some dimension information attached to it could be used as an argument to those functions, and thus we would not need to update these type aliases every time we implement some new class or decide to re-structure the class. And I think this is what is done in `sympy` to some extent.

Originally, I just wanted `Wrapper` to be a subclass of `DimensionSymbolNew` (since `Wrapper`s are semantically closer to `Symbol`s and they carry some dimension information), but that would mean I would have to set the `_display_name` and `_display_latex` fields, which is sort of weird since that is implemented in `symplyphysics/docs/printer_*.py` modules.